oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java

```diff
@@ -104,6 +104,10 @@ public static boolean isDayBucket(long timeBucket) {
     public static long getTimestamp(long timeBucket, DownSampling downsampling) {
         Calendar calendar = Calendar.getInstance();
         calendar.setTimeInMillis(0);
+        calendar.set(Calendar.MILLISECOND, 0);
+        calendar.set(Calendar.SECOND, 0);
+        calendar.set(Calendar.MINUTE, 0);
+        calendar.set(Calendar.HOUR_OF_DAY, 0);
```

Comment:
Do you need these if you already have `calendar.setTimeInMillis(0);`?

Reply:
Yes, they are required. The time equals zone offset  when `calendar.setTimeInMillis(0);` was presented. And then, I found some time zone is not a times as hour. But they all is a times as minute. (L107/L108 is optional.)